src/ray/core_worker/core_worker.cc

```diff
@@ -983,6 +1018,20 @@ Status CoreWorker::ExecuteTask(const TaskSpecification &task_spec,
     }
   }
 
+  // Get the reference counts for any IDs that we borrowed during this task and
+  // return them to the caller. This will notify the caller of any IDs that we
+  // (or a nested task) are still borrowing. It will also any new IDs that were
+  // contained in a borrowed ID that we (or a nested task) are now borrowing.
+  reference_counter_->GetAndStripBorrowedRefs(borrowed_ids, borrowed_refs);
+  // Unpin the borrowed IDs.
+  std::vector<ObjectID> deleted;
+  for (const auto &borrowed_id : borrowed_ids) {
+    reference_counter_->RemoveLocalReference(borrowed_id, &deleted);
+  }
+  if (ref_counting_enabled_) {
+    memory_store_->Delete(deleted);
```

Comment:
Do we ever put any borrowed objects in the memory store? I was under the impression the memory store only holds return objects from tasks we own.

Reply:
We do, but it's just to put a plasma error ([here](https://github.com/ray-project/ray/blob/master/src/ray/core_worker/future_resolver.cc#L29)).

Reply:
Add a comment here to this effect?